static/js/settings_display.js

```diff
@@ -96,38 +95,55 @@ export function set_up(settings_panel) {
         }
     });
 
-    $("#user_default_language_modal")
-        .find(".language")
-        .on("click", (e) => {
-            e.preventDefault();
-            e.stopPropagation();
-            overlays.close_modal("#user_default_language_modal");
-
-            const $link = $(e.target).closest("a[data-code]");
-            const setting_value = $link.attr("data-code");
-            const data = {default_language: setting_value};
-
-            const new_language = $link.attr("data-name");
-            container.find(".default_language_name").text(new_language);
-
-            change_display_setting(
-                data,
-                container.find(".language-settings-status"),
-                $t_html(
-                    {
-                        defaultMessage:
-                            "Saved. Please <z-link>reload</z-link> for the change to take effect.",
-                    },
-                    {"z-link": (content_html) => `<a class='reload_link'>${content_html}</a>`},
-                ),
-                true,
-            );
-        });
+    function default_language_modal_post_render() {
+        $("#user_default_language_modal")
+            .find(".language")
+            .on("click", (e) => {
+                e.preventDefault();
+                e.stopPropagation();
+                dialog_widget.close_modal();
+
+                const $link = $(e.target).closest("a[data-code]");
+                const setting_value = $link.attr("data-code");
+                const data = {default_language: setting_value};
+
+                const new_language = $link.attr("data-name");
+                container.find(".default_language_name").text(new_language);
+
+                change_display_setting(
+                    data,
+                    container.find(".language-settings-status"),
+                    $t_html(
+                        {
+                            defaultMessage:
+                                "Saved. Please <z-link>reload</z-link> for the change to take effect.",
+                        },
+                        {"z-link": (content_html) => `<a class='reload_link'>${content_html}</a>`},
+                    ),
+                    true,
+                );
+            });
+    }
 
     container.find(".setting_default_language").on("click", (e) => {
         e.preventDefault();
         e.stopPropagation();
-        overlays.open_modal("#user_default_language_modal");
+
+        const html_body = render_dialog_default_language({
+            language_list: get_language_list_columns(page_params.default_language),
```

Comment:
This should be `user_settings.default_language` not `page_params.default_langauge` which is the realm-level default.

Reply:
Done